README.md

```diff
@@ -447,6 +448,82 @@ func main() {
 }
 ```
 
+### Middleware
+
```

Comment:
* Add a quick paragraph describing what middleware is / what it can be used for, and link to https://github.com/gorilla/handlers
* Version will be v1.6.1 if we merge this week (I'll cut a new release after this PR is merged in)

e.g.

> Middleware describes intermediate handlers that can check, mutate or log the incoming request or outgoing response. mux makes it easy to add middleware to a [Router](https://godoc.org/github.com/gorilla/mux#Router) or Subrouter...

Reply:
Added!

Reply:
Btw, I forgot to mention I did not add the link because most handler don't actually implement `mux.Middleware`, as they require additional parameters. We could circunvent this using closures, but I think it would confuse some users instead of adding clarity. Maybe we could rework some handlers to make them comply with the interface?

Reply:
Maybe instead of changing them we could add a `XXXXMiddleware` function for each one?

Reply:
Yes, of course. We should not break existing handlers, so maybe wrapping them somehow, or refactoring the core logic into a different, private function and exposing both the actual and the Middleware flavours of it. I can give it a shot if you think it's worth it.